src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -288,11 +303,35 @@ def _max_by_axis(self, the_list: List[List[int]]) -> List[int]:
                 maxes[index] = max(maxes[index], item)
         return maxes
 
+    def convert_segmentation_map_to_binary_masks(self, segmentation_map: "np.ndarray", num_labels: int):
+        # get all the labels in the image
+        labels = np.unique(segmentation_map)
+        if self.reduce_labels:
+            segmentation_map[segmentation_map == 0] = 255
+            segmentation_map = segmentation_map - 1
+            segmentation_map[segmentation_map == 254] = self.ignore_index
+        # remove ignore index (if we have one)
+        if self.ignore_index is not None:
+            labels = labels[labels != self.ignore_index]
```

Comment:
Don't think this is correct, you are first replacing labels by 255 (or another ignore index), and then you are just checking the original labels?

Reply:
correct, updated! (thanks :pray: )

Reply:
nice catch, updated